src/uct/ib/base/ib_iface.c

```diff
@@ -245,59 +245,65 @@ size_t uct_ib_address_size(uct_ib_iface_t *iface)
     }
 }
 
-void uct_ib_address_pack(uct_ib_iface_t *iface,
-                         const union ibv_gid *gid, uint16_t lid,
+size_t uct_ib_iface_address_size(uct_ib_iface_t *iface)
+{
+    return uct_ib_addr_size(&iface->gid, iface->is_global_addr, uct_ib_iface_is_roce(iface));
+}
+
+void uct_ib_address_pack(const union ibv_gid *gid, uint16_t lid,
```

Comment:
why need to separate setting the flags and packing the address?
maybe just pass add needed parameters to pack function, instead of passing iface handle?